### PR TITLE
Dont place ents in eye cell

### LINKED ISSUE #148: Frobnicator can be placed in eye cell, and afterward the player is stuck forever.
We should just ban this. We probably also need to either frag the player or magically unstick them if we find them inside an entity.

### PR DESCRIPTION
**Pull request by chrisforbes**
Fixes #148

### CODE REVIEW

File: main.cc [hunk 1]
```diff
@@ -92,6 +92,15 @@ light_field *light;
 entity *use_entity = nullptr;
 
 
+glm::ivec3
+get_block_containing(glm::vec3 v) {
```
Comment (mkfifo) [inline main.cc]: this doesn't return a block.

File: main.cc [hunk 2]
```diff
@@ -601,6 +610,12 @@ struct add_block_entity_tool : tool
         if (!rc->hit || rc->inside)
             return false;
 
+        /* don't allow placements that would cause the player to end up inside the ent and get stuck */
```
Comment (mkfifo) [inline main.cc]: glad we have captured this case, the other day playing I was trying to make it happen and (by luck it seems) wasn't able to trap myself.

**Comment by mkfifo** [comment]
can't test until home, looks good otherwise.

File: main.cc (hunk 1 above)
Comment (chrisforbes) [inline main.cc]: ok, what do we want to call it?

block coord?

File: main.cc (hunk 2 above)
Comment (chrisforbes) [inline main.cc]: The easy way was to crouch, and aim at the floor.

File: main.cc (hunk 1 above)
Comment (mkfifo) [inline main.cc]: most accurate I could come up with was 'get_block_coord_containing', so 'get_block_coords' works.

**Comment by chrisforbes** [comment]
ping

**Comment by mkfifo** [comment]
Confirmed this fixes the issue, I can no longer place myself inside a frobnicator.

**Pull request merged**